code/http4s/src/main/scala/com/avast/datadog4s/extension/http4s/impl/DefaultMetricsOps.scala

```diff
@@ -15,10 +16,29 @@ private[http4s] class DefaultMetricsOps[F[_]](
     metricFactory: MetricFactory[F],
     classifierTags: ClassifierTags,
     activeConnectionsRef: Ref[F, ActiveConnections],
-    distributionBasedTimers: Boolean
+    distributionBasedTimers: Boolean,
+    distributionBasedCounters: Boolean
 )(implicit
     F: Sync[F]
 ) extends MetricsOps[F] {
+
+  private[http4s] sealed trait GenericCounter {
+    def inc(tags: Tag*): F[Unit]
+  }
+
+  private[http4s] object GenericCounter {
+    final class CountWrapper(aspect: String) extends GenericCounter {
+      private val counter = metricFactory.count(aspect)
+
+      override def inc(tags: Tag*): F[Unit] = counter.inc(tags*)
+    }
+    final class DistributionCounter(aspect: String) extends GenericCounter {
+      private val counter = metricFactory.distribution.long(aspect)
+
+      override def inc(tags: Tag*): F[Unit] = counter.record(1L, tags*)
+    }
+  }
```

Comment:
alright, so this is the only thing i would change - please move this to it's own file in order to make it more testable and keep this class shorter and therefore understandable. Other than that i think we can merge this.

Reply:
This is done!

Reply:
@tomasherman How does this look to you?

Reply:
lgtm, lets do it :) Sorry its been busy few weeks in a new job :)